pkg/configprofiles/profiles.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package configprofiles
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/clusterversion"
+	"github.com/cockroachdb/cockroach/pkg/server/autoconfig/autoconfigpb"
+)
+
+// alias represents a name that points to a pre-defined profile with
+// an alternate description.
+type alias struct {
+	aliasTarget string
+	description string
+}
+
+var aliases = map[string]alias{
+	"replication-source": {
+		aliasTarget: "multitenant+app+sharedservice",
+		description: "configuration suitable for a replication source cluster",
+	},
+	"replication-target": {
+		aliasTarget: "multitenant+noapp",
+		description: "configuration suitable for a replication target cluster",
+	},
+}
+
+const defaultProfileName = "default"
+
+type configProfile struct {
+	description string
+	tasks       []autoconfigpb.Task
+}
+
+var staticProfiles = map[string]configProfile{
+	defaultProfileName: {
+		// Do not define tasks in the "default" profile. It should continue
+		// to generate clusters with default configurations.
+		description: "no extra configuration applied - using source code defaults",
+	},
+	// The "example" profile exists for demonstration and documentation
+	// purposes.
+	"example": {
+		description: "creates an 'example' database and data table, for illustration purposes",
+		tasks: []autoconfigpb.Task{
+			makeTask("create an example database",
+				/* nonTxnSQL */ nil,
+				/* txnSQL */ []string{
+					"CREATE DATABASE IF NOT EXISTS example",
+					"CREATE TABLE IF NOT EXISTS example.data AS SELECT 'hello' AS value",
+				},
+			),
+		},
+	},
+	"multitenant+noapp": {
+		description: "multi-tenant cluster with no secondary tenant defined yet",
+		tasks:       multitenantClusterInitTasks,
+	},
+	"multitenant+app+sharedservice": {
+		description: "multi-tenant cluster with one secondary tenant configured to serve SQL application traffic",
+		tasks: append(
+			multitenantClusterInitTasks,
+			makeTask("create an application tenant",
+				nil, /* nonTxnSQL */
+				/* txnSQL */ []string{
+					// Create the app tenant record.
+					"CREATE TENANT application",
+					// Run the service for the application tenant.
+					"ALTER TENANT application START SERVICE SHARED",
+				},
+			),
+			makeTask("activate application tenant",
+				/* nonTxnSQL */ []string{
+					// Make the app tenant receive SQL connections by default.
+					"SET CLUSTER SETTING server.controller.default_tenant = 'application'",
+				},
+				nil, /* txnSQL */
+			),
+		),
+	},
+}
+
+var multitenantClusterInitTasks = []autoconfigpb.Task{
+	makeTask("initial cluster config",
```

Comment:
We could also enable rangefeed's here so that it can be used as a replication source more easily.